vortex-array/src/compute/fill_null.rs

```diff
@@ -0,0 +1,49 @@
+use vortex_error::{vortex_bail, vortex_err, VortexError, VortexResult};
+use vortex_scalar::Scalar;
+
+use crate::encoding::Encoding;
+use crate::{ArrayDType, ArrayData};
+
+/// Implementation of fill_null for an encoding.
+///
+/// SAFETY: the fill value is guaranteed to be non-null.
```

Comment:
```suggestion
/// An error is returned if the `fill_value` is `null`.
```

Reply:
The top level function checks this. At this level you can assert its non null